jumpscale/packages/tfgrid_solutions/chats/discourse.py

```diff
@@ -0,0 +1,243 @@
+        self.solution_metadata = {}
+
+    @chatflow_step(title="Solution name")
+    def discourse_name(self):
+        self.solution_name = deployer.ask_name(self)
+
+    @chatflow_step(title="SMTP server")
+    def discourse_smtp_server(self):
+        self.smtp_server = self.string_ask("Please, enter the smtp server address", required=True)
+
+    @chatflow_step(title="SMTP username")
+    def discourse_smtp_username(self):
+        self.smtp_username = self.string_ask("Please, enter the smtp server username", required=True)
+
+    @chatflow_step(title="SMTP password")
+    def discourse_smtp_password(self):
+        self.smtp_password = self.string_ask("Please, enter the smtp server password", required=True)
+
+    @chatflow_step(title="Email")
+    def discourse_email(self):
+        self.email = deployer.ask_email(self)
+        self.container_resources()
+
+    def container_resources(self):
+        self.resources = dict()
+        self.resources["cpu"] = 1
+        self.resources["memory"] = 2048
+        self.resources["disk_size"] = 4096
+        self.resources["default_disk_type"] = "SSD"
+
+    @chatflow_step(title="Pool")
+    def select_pool(self):
+        query = {
+            "cru": self.resources["cpu"],
+            "mru": math.ceil(self.resources["memory"] / 1024),
+            "sru": math.ceil(self.resources["disk_size"] / 1024),
+        }
+        cu, su = deployer.calculate_capacity_units(**query)
+        self.pool_id = deployer.select_pool(self, cu=cu, su=su)
+
+    @chatflow_step(title="Network")
+    def discourse_network(self):
+        self.network_view = deployer.select_network(self)
+        self.container_logs()
+        self.container_node_id()
+        self.container_ip()
+        self.select_domain()
+
+    def container_logs(self):
+        self.log_config = {}
+
+    def container_node_id(self):
+        query = {
+            "cru": self.resources["cpu"],
+            "mru": math.ceil(self.resources["memory"] / 1024),
+            "sru": math.ceil(self.resources["disk_size"] / 1024),
+        }
+        self.selected_node = deployer.schedule_container(self.pool_id, **query)
+
+    def container_ip(self):
+        self.network_view_copy = self.network_view.copy()
+        result = deployer.add_network_node(
+            self.network_view.name, self.selected_node, self.pool_id, self.network_view_copy, bot=self
+        )
+        if result:
+            self.md_show_update("Deploying Network on Nodes....")
+            for wid in result["ids"]:
+                success = deployer.wait_workload(wid)
+                if not success:
+                    raise StopChatFlow(f"Failed to add node {self.selected_node.node_id} to network {wid}")
+            self.network_view_copy = self.network_view_copy.copy()
+        self.ip_address = self.network_view_copy.get_free_ip(self.selected_node)
+
+    def select_domain(self):
+        self.gateways = {
```

Comment:
i think this may cause a problem with `FreeTFT` gateways

maybe we can use

```python
        self.gateways = {}
        for g in filter(j.sals.zos.nodes_finder.filter_is_up, j.sals.zos._explorer.gateway.list()):
            if self.network.currency == "FreeTFT" and not g.free_to_use:
                continue
            self.gateways[g.node_id] = g
```